# House Prices Attempt I: Linear Regression

For my first non-physics machine learning project I will be testing out various techniques on a straightforward house prices dataset. The first attempt will be a simple linear regression. The plan is to use one hot encoding (etc.) to deal with the categorical data which should be easy to do with the **pandas** toolset. I also expect some ordinal features will not lend well to a simple linear model, so I will also try making up some nonlinear features (*e.g.* (Overall Quality)^2).  I expect that the number of features may become unwieldly, but we will see :).

Standard imports:

In [3]:
%matplotlib inline
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()
import numpy as np
import pandas as pd

Load raw training data:

In [51]:
data_train = pd.read_csv("../data/train.csv")
print(data_train.shape)

(1460, 81)


OK, so there are $m = 1460$ training examples and $n = 80$ features (ignoring 'Id'). So I'm pretty sure after one hot encoding $n$ will be approaching $m$. This may be bad but I'm going to allow it just to see what happens.

## One hot encoding
Here are the categorical features that I have decided to encode:

In [65]:
ohe_categories = [
    'MSSubClass',
    'MSZoning',
    'Street',
    'Alley',
    'LotShape',
    'LandContour',
    'LotConfig',
    'Neighborhood',
    'Condition1',
    'Condition2',
    'BldgType',
    'HouseStyle',
    'RoofStyle',
    'RoofMatl',
    'Exterior1st',
    'Exterior2nd',
    'MasVnrType',
    'Foundation',
    'Heating',
    'Electrical',
    'GarageType',
    'MiscFeature',
    'SaleType',
    'SaleCondition'
]
data_augmented = pd.get_dummies(data_train[ohe_categories])
data_augmented[:3]

MSSubClass  MSZoning_C (all)  MSZoning_FV  MSZoning_RH  MSZoning_RL  \
0          60                 0            0            0            1   
1          20                 0            0            0            1   
2          60                 0            0            0            1   

   MSZoning_RM  Street_Grvl  Street_Pave  Alley_Grvl  Alley_Pave  \
0            0            0            1           0           0   
1            0            0            1           0           0   
2            0            0            1           0           0   

           ...            SaleType_ConLw  SaleType_New  SaleType_Oth  \
0          ...                         0             0             0   
1          ...                         0             0             0   
2          ...                         0             0             0   

   SaleType_WD  SaleCondition_Abnorml  SaleCondition_AdjLand  \
0            1                      0                      0   
1            1                      0                      0   
2            1                      0                      0   

   SaleCondition_Alloca  SaleCondition_Family  SaleCondition_Normal  \
0                     0                     0                     1   
1                     0                     0                     1   
2                     0                     0                     1   

   SaleCondition_Partial  
0                      0  
1                      0  
2                      0  

[3 rows x 169 columns]

I will add the treated data to the *data_augmented* variable as I go.